docs(grid): Enhance Grid ExpandoObject KB (#1370)
telerik/blazor-docs: knowledge-base/grid-binding-to-expando-object.md

* If you need [autogenerated Grid columns]({%slug grid-columns-automatically-generated%}), then [define them in a loop inside the `<GridColumns>` tag]({%slug grid-kb-dynamic-columns-with-static%}), which is a standard Blazor `RenderFragment`.
* If the Grid columns are created from the keys of *first* data item, they will disappear when adding a new item. This is because the newly added data item (`ExpandoObject`) has no properties and is *prepended* to the `Data` collection. To avoid this, either create columns from a non-first item, or use [`OnModelInit`]({%slug grid-events%}#onmodelinit) to populate the newly added item with some default values. `null` does not qualify as a default value, because it cannot help determine the property type. The example below includes an `OnModelInit` handler.
* The Grid clones data items during editing. When editing, you can store the original item in the [Grid `OnEdit` handler]({%slug components/grid/editing/overview%}#events). This will make the retrieval of the original item easier later in `OnUpdate` (example below).
* Nested model properties and nullable types need special handling. Make sure to [check the notes](#notes) below the example.

> In addition, there is a [complete runnable project in GitHub](https://github.com/telerik/blazor-ui/tree/master/grid/binding-to-expando-object).

Reviewer: In addition to what?
```suggestion
See the [complete runnable project in GitHub](https://github.com/telerik/blazor-ui/tree/master/grid/binding-to-expando-object).
```

> In addition, there is a [complete runnable project in GitHub](https://github.com/telerik/blazor-ui/tree/master/grid/binding-to-expando-object).

Author: ```suggestion
In addition to the sample below, there is a [complete runnable project in GitHub](https://github.com/telerik/blazor-ui/tree/master/grid/binding-to-expando-object).
```

````

## Notes

> * It is possible to use [nested Grid model properties]({%slug grid-use-navigation-properties%}) that are `ExpandoObject`s. However, data operations like filtering, sorting and grouping are not supported for those nested properties. [Disable these features per column]({%slug components/grid/columns/bound%}#data-operations) if they are enabled for the Grid.

Reviewer: ```suggestion
* You can use [nested Grid model properties]({%slug grid-use-navigation-properties%}) that are `ExpandoObject`s. However, data operations like filtering, sorting, and grouping are not supported for those nested properties. [Disable these features per column]({%slug components/grid/columns/bound%}#data-operations) if they are enabled for the Grid.
```

Author: ```suggestion
* You can also use [nested Grid model properties]({%slug grid-use-navigation-properties%}) that are `ExpandoObject`s. However, data operations like filtering, sorting and grouping are not supported for those nested properties. [Disable these features per column]({%slug components/grid/columns/bound%}#data-operations) if they are enabled for the Grid.
```

> * If any property in the `ExpandoObject` is `nullable`, then do not set [`EditorType`]({%slug components/grid/editing/overview%}#customize-the-editor-fields) for the column (this may be supported in the future). Instead, use a [Grid column `EditorTemplate`](grid-templates-editor), for example:

Author: ```suggestion
* If any property in the `ExpandoObject` is `nullable`, then do not set [`EditorType`]({%slug components/grid/editing/overview%}#customize-the-editor-fields) for the column (this may be supported in the future). Instead, use a [Grid column `EditorTemplate`]({%slug grid-templates-editor%}), for example:
```
